### ISSUE TITLE
Commit Graph: Pull-Requests should not link to issues

### ISSUE THREAD

**Issue by markusamshove** [go-gitea/gitea#15730]
- Gitea version (or commit ref): 1.14.1

## Description

With Gitea 1.14.0 and #12766 the commit graph (`<repository-url>/graph`)  includes links to pull requests instead of the names of the ref's.

I like this change a lot, however I can't really use it, because my repositories use an external issue tracker.

The links associated with the pull requests are generated as: `<repository-url>/issues/<pr-id>`
The links would work if they were generated as: `<repository-url>/pulls/<pr-id>`

**Comment by zeripath** [comment]
I'm confused because these should have already been being redirected if Gitea was set up correctly.

**Comment by MarkusAmshove** [comment]
What do I have to configure in Gitea other than the external tracker in the repository?

If I configure an external tracker, e.g. Redmine, all pull reference links in Gitea (like the automated commit message in merges and now the graph) always redirect to Redmine when clicked, because they use the issue route and not the pulls route.

If changing those links to pulls is no option, then they shouldn't be generated as links in the first place when an external tracker is configured, because it is confusing

**Comment by MarkusAmshove** [comment]
By the way, the external tracker uses the same format as Gitea uses for issues/pr: `#123`

That's why all generated Gitea messages which reference pulls also redirect to the external tracker

**Comment by zeripath** [comment]
agh crap - now I remember - the external issue tracker vs external pull requests thing is not completely specified.

I think we need to pull in @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf here. How have you configured this?

There has to be an explicit way of detecting what we should be linking to.

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
the thing is that *sometimes* you might actually want to link both issues and PRs externally - such as for a mirror.
you don't track issues for the mirror, much less PRs, you specify the external issue tracker (known format for GH, another Gitea instance, Gitlab...) and PRs get autolinked outside as well.
that's what I do for mirrors I set up, it's makes for a very convenient navigation flow.

but that's just one use case.

another one might be the one OP mentions, where issues are tracked externally but PR's are local.
to satisfy them both it might be smart to have a way of specifying whether it's desirable to also externally link PRs when external issue tracker is enabled.
it could, e.g., be unchecked by default for a mirror (no reason to track PRs) and this checkbox would only be available if external issue tracker is enabled (cause otherwise it makes no sense).

to summarize, I have my own way of using the feature but I recognize there's other uses for it, such as the one mentioned by OP.

**Comment by zeripath** [comment]
@wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf do your repositories here have the pull requests feature turned off or still turned on?

Similarly @MarkusAmshove ?

It would be good to see a screenshot of the repository advanced settings sections, eg.:

![Screenshot from 2021-05-06 22-23-04](https://user-images.githubusercontent.com/1824502/117367139-ab3cbf80-aeb9-11eb-8a10-b2967d92be7e.png)

(Clearly censored as necessary.)

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
here's mine for a github.com/go-gitea/gitea mirror
![image](https://user-images.githubusercontent.com/61180606/117374431-6ae73c80-aecd-11eb-968e-21c4edaff456.png)

![image](https://user-images.githubusercontent.com/61180606/117374535-a1bd5280-aecd-11eb-9204-1199e9b7df16.png)

**Comment by zeripath** [comment]
@MarkusAmshove I guess you have pullrequests enabled? If so that hunts at how the square may be circled and we can get a PR that solves both issues

**Comment by MarkusAmshove** [comment]
> @MarkusAmshove I guess you have pullrequests enabled? If so that hunts at how the square may be circled and we can get a PR that solves both issues

I can't provide a screenshot yet, but yes:

- External issue tracker enabled 
- Pull Requests enabled

**Comment by zeripath** [comment]
Cool there exists a solution that will make both situations work.
